vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt

```diff
@@ -95,6 +96,23 @@ class UiAllScreensSanityTest {
             }
         }
 
+        elementRobot.space {
```

Comment:
for some more context about the robot pattern, it aims to create page models for each of the pages/screens/interactable components and return to a consistent state after usage

it allows for building and combining flows together like 

```
elementRobot.settings {
    advancedSettings { crawl() }
}

elementRobot.withDeveloperMode {
    settings {
        advancedSettings { crawlDeveloperOptions() }
    }
}
```
 
with that in mind there's some minor changes we could make to the spaces to make them easier to reuse

- parameterised space creation
- robot for the space menu (with ability to reopen itself)

```kotlin
elementRobot.space {
    val spaceName = UUID.randomUUID().toString()
    createSpace {
        crawl()
        createPublicSpace(spaceName)
    }
    spaceMenu(spaceName) {
        listMembers()
        settings()
        exploreRooms()

        // could be internal to the interactions that close the menu
        keepMenuOpen {
            invitePeople()
            addRoom()
            addSpace()
            leaveSpace()
        }
    }
}
```

what do you think?

Reply:
(maybe those useful information could be paste in the documentation of the project somewhere)

Reply:
very good point :+1:  I'll update the UI tests doc

Reply:
added https://github.com/vector-im/element-android/pull/5228